File: data/src/main/java/com/example/electricalcalculator/data/db/entity/ApplianceEntity.java
package com.example.electricalcalculator.data.db.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * Entity representation of an appliance in the database
 */
@Entity(tableName = "appliances")
public class ApplianceEntity {
    
    @PrimaryKey(autoGenerate = true)
    private long id;
    
    private String name;
    private double wattage;
    private double voltageRating;
    private boolean isMotorLoad;

    public ApplianceEntity(String name, double wattage, double voltageRating, boolean isMotorLoad) {
        this.name = name;
        this.wattage = wattage;
        this.voltageRating = voltageRating;
        this.isMotorLoad = isMotorLoad;
    }

    // Getters and setters
    public long getId() { return id; }
    public void setId(long id) { this.id = id; }
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    
    public double getWattage() { return wattage; }
    public void setWattage(double wattage) { this.wattage = wattage; }
    
    public double getVoltageRating() { return voltageRating; }
    public void setVoltageRating(double voltageRating) { this.voltageRating = voltageRating; }
    
    public boolean isMotorLoad() { return isMotorLoad; }
    public void setMotorLoad(boolean motorLoad) { isMotorLoad = motorLoad; }
}

File: app/src/main/java/com/example/electricianapp/data/local/entity/ApplianceEntity.kt
package com.example.electricianapp.data.local.entity // Corrected package

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Entity representation of an appliance in the database.
 * This seems related to a more detailed appliance database, potentially
 * distinct from the simple Appliance model used in DwellingLoadInput.
 * Adjust fields as necessary based on actual requirements.
 */
@Entity(tableName = "appliances") // Define table name
data class ApplianceEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,

    val name: String,
    val wattage: Double,
    val voltageRating: Double?, // Voltage might be optional
    val isMotorLoad: Boolean = false // Default to false if not specified

    // Consider adding fields like:
    // val phase: String? // e.g., SINGLE_PHASE, THREE_PHASE (use enum + TypeConverter)
    // val description: String?
    // val category: String? // e.g., Kitchen, HVAC, Motor
)
